docs/design/mp_arktos.md

```diff
@@ -0,0 +1,229 @@
+-->
+
+# Arktos Network and Mizar Integration
+
+## Introduction
+
+This document is designated for Arktos service using Mizar as its underline network system. In this document, you will find the design detail on how to integration Arktos multi-tenancy network object with Mizar as its network platform, so that Arktos can fully utilize network objects offered in Mizar, including vpcs, nets, services, pods, bouncers and dividers. 
+
+## Arktos Network Object Overview: 
+The Arktos network object contains its type and type-specific configurations, which translates into spec fields type and vpcID:
+```yaml
+apiVersion: arktos.futurewei.com/v1
+kind: Network
+metadata:
+  name: vpc-1
+spec:
+  type: mizar
+  vpcID: vpc-1a2b3c4d
+```
+The type field is the only required field, where it identifies the underline network service (which in our case is mizar). VpcID field is optional, and it lists the vpc id that is associated with this Arktos network object. If vpcID is not specified, the default VPC will be used. 
+
+Below is the definition of a flat network, where the defaul network is used: 
+```yaml
+apiVersion: arktos.futurewei.com/v1
+kind: Network
+metadata:
+  name: default
+spec:
+  type: flat
+```
+
+The Arktos network object and VPC have **many to one relationship**, where multiple Arktos network objects can reference the same VPC network. 
+
+## Pod Definition
+When a pod is attached to the default network, nothing is needed in pod spec:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    ports:
+      - containerPort: 443
+```
+
+When a pod is attached to a certain network, it needs to set its "network" in labels:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    arktos.futurewei.com/network: vpc-1
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    ports:
+      - containerPort: 443
+```
+
+When a pod is attached to a certain network, then it needs to specify subnet or IP. It can be expressed in annotations:
+
+```yaml
+
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    arktos.futurewei.com/network: vpc-1
+  annotations:
+    arktos.futurewei.com/nic: {"name": "eth0", "ip": "10.10.1.12", "subnet": "net1"}
+    arktos.futurewei.com/vip-hint: {"cidr":"10.10.1.0/26", "ip":"10.10.1.12", "policy":"BestEffort"}
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    ports:
+      - containerPort: 443
+
+```
+
+Annotation ```arktos.futurewei.com/nic``` is for user to provide optional information about pod nic. The recognized keys of element include:
+
+|key|exemplary value|
+|---:|---:|
+|name|eth0|
+|subnet|subnet-1, or 192.168.100.0/26|
+|ip|192.168.100.5|
+
+Please note that any of the three fields can be missing. For example, user might only specify  ```ip``` field; in this case, Mizar needs to figure out which subnet this ip falls into, and creates the pod into the correct subnet. 
```

Comment:
Will a subnet be created if no pre-existing subnet contains the IP address in its range?

Reply:
For version one, we do not support any new object creation on Mizar side. If no pre-existing subnet, mizar will throw an error